internal/hydra/device.go

```diff
@@ -0,0 +1,231 @@
+package hydra
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	hClient "github.com/ory/hydra-client-go/v2"
+)
+
+// We implement the device API logic, as the upstream sdk does not support it.
+// Otherwise we would have to fork the sdk
+// TODO(nsklikas): Remove this once upstream hydra supports the device flow
+
+type DeviceApiService struct {
+	client *hClient.APIClient
+	hClient.OAuth2Api
+}
+
+type APIError struct {
+	body  []byte
+	error string
+}
+
+// Error returns non-empty string if there was an error.
+func (e APIError) Error() string {
+	return e.error
+}
+
+// Body returns the raw bytes of the response
+func (e APIError) Body() []byte {
+	return e.body
+}
+
+// Prevent trying to import "fmt"
+func reportError(format string, a ...interface{}) error {
+	return fmt.Errorf(format, a...)
+}
+
+// AcceptDeviceUserCodeRequest Contains information on an device verification
+type AcceptDeviceUserCodeRequest struct {
+	UserCode *string `json:"user_code,omitempty"`
```

Comment:
do we need validation here?

Reply:
This struct eventually will be provided by the ory sdk, which does not provide validation. So I think we should leave it as it is and if we want for further validation we should implement it at another layer.